src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -141,7 +151,29 @@ public void shoot() {
     } else {
       shooterGate.set(0);
     }
+  }
+
+  public void shootFromWall() {
+    // fast flywheel
+    // deploy hood at lower angle
+    // run gate if both of "" are good
+    double openLoopSetpoint = Utils.map(MoPrefs.getShooterPIDSetpoint(), -5500, 5500, -1, 1);
```

Comment:
This pattern has appeared multiple times.  Please refactor into a single function that eliminates duplicated code.  Also do something about the magic numbers.